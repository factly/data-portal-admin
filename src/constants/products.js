export const LOADING_PRODUCTS = 'LOADING_PRODUCTS';
export const LOAD_PRODUCTS_SUCCESS = 'LOAD_PRODUCTS_SUCCESS';
export const ADD_PRODUCTS_LIST_REQUEST = 'ADD_PRODUCTS_LIST_REQUEST';
export const SET_PRODUCTS_LIST_CURRENT_PAGE = 'SET_PRODUCTS_LIST_CURRENT_PAGE';
export const SET_PRODUCTS_LIST_TOTAL = 'SET_PRODUCTS_LIST_TOTAL';
export const LOAD_PRODUCTS_FAILURE = 'LOAD_PRODUCTS_FAILURE';
export const CREATING_PRODUCT = 'CREATING_PRODUCT';
export const CREATE_PRODUCT_SUCCESS = 'CREATE_PRODUCT_SUCCESS';
export const CREATE_PRODUCT_FAILURE = 'CREATE_PRODUCT_FAILURE';
export const LOADING_PRODUCT_DETAILS = 'LOADING_PRODUCT_DETAILS';
export const GET_PRODUCT_DETAILS_SUCCESS = 'GET_PRODUCT_DETAILS_SUCCESS';
export const GET_PRODUCT_DETAILS_FAILURE = 'GET_PRODUCT_DETAILS_FAILURE';
export const UPDATING_PRODUCT = 'UPDATING_PRODUCT';
export const UPDATE_PRODUCT_SUCCESS = 'UPDATE_PRODUCT_SUCCESS';
export const UPDATE_PRODUCT_FAILURE = 'UPDATE_PRODUCT_FAILURE';
export const DELETING_PRODUCT = 'DELETING_PRODUCT';
export const DELETE_PRODUCT_SUCCESS = 'DELETE_PRODUCT_SUCCESS';
export const DELETE_PRODUCT_FAILURE = 'DELETE_PRODUCT_FAILURE';

export const baseUrl = `${process.env.REACT_APP_API_URL}/products`;
