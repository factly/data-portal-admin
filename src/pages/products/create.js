import React, { useState } from 'react';
import { Form, Input, Button, notification, Select } from 'antd';

const { Option } = Select;

const formItemLayout = {
  labelCol: { span: 6 },
  wrapperCol: { span: 14 },
};

const ProductCreate = (props) => {
  const [tags, setTags] = useState([]);
  const [categories, setCategories] = useState([]);
  const [currencies, setCurrencies] = useState([]);
  const [productType, setProductType] = useState([]);

  React.useEffect(() => {
    fetch(process.env.REACT_APP_API_URL + '/tags')
      .then((data) => data.json())
      .then((data) => {
        setTags(data.nodes);
      });
    fetch(process.env.REACT_APP_API_URL + '/categories')
      .then((data) => data.json())
      .then((data) => {
        setCategories(data.nodes);
      });
    fetch(process.env.REACT_APP_API_URL + '/currencies')
      .then((data) => data.json())
      .then((data) => {
        setCurrencies(data.nodes);
      });
    fetch(process.env.REACT_APP_API_URL + '/types')
      .then((data) => data.json())
      .then((data) => {
        setProductType(data.nodes);
      });
  }, []);

  function handleChange(value) {
    console.log(`selected ${value}`);
  }

  const onFinish = (values) => {
    values.currency_id = parseInt(values.currency_id);
    values.product_type_id = parseInt(values.product_type_id);
    values.price = parseInt(values.price);
    values.category_ids = values.category_ids.map((c) => parseInt(c));
    values.tag_ids = values.tag_ids.map((t) => parseInt(t));
    fetch(process.env.REACT_APP_API_URL + '/products', {
      method: 'POST',
      body: JSON.stringify(values),
    })
      .then((res) => {
        if (res.status === 201) {
          return res.json();
        } else {
          throw new Error(res.status);
        }
      })
      .then((_) => {
        notification.success({
          message: 'Success',
          description: 'Product succesfully added',
        });
        props.history.push('/products');
      })
      .catch((res) => {
        notification.error({
          message: 'Error',
          description: 'Something went wrong',
        });
      });
  };

  return (
    <Form name="products_create" {...formItemLayout} onFinish={onFinish}>
      <Form.Item
        label="Title"
        name="title"
        rules={[
          {
            required: true,
            message: 'Please enter title!',
          },
        ]}
      >
        <Input placeholder="Ex. Crime In India" />
      </Form.Item>

      <Form.Item
        label="Slug"
        name="slug"
        rules={[
          {
            required: true,
            message: 'Please enter slug!',
          },
        ]}
      >
        <Input placeholder="Ex. crime-in-india" />
      </Form.Item>

      <Form.Item
        label="Currency"
        name="currency_id"
        rules={[
          {
            required: true,
            message: 'Please enter currency!',
          },
        ]}
      >
        <Select
          showSearch
          style={{ width: '100%' }}
          placeholder="Select currency"
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          {currencies.length > 0
            ? currencies.map((c) => <Option key={c.id}>{c.iso_code}</Option>)
            : []}
        </Select>
      </Form.Item>

      <Form.Item
        label="Price"
        name="price"
        rules={[
          {
            required: true,
            message: 'Please enter price!',
          },
        ]}
      >
        <Input placeholder="Ex. 1999" />
      </Form.Item>

      <Form.Item
        label="Category"
        name="category_ids"
        rules={[
          {
            required: true,
            message: 'Please select atleast one category!',
          },
        ]}
      >
        <Select
          mode="tags"
          style={{ width: '100%' }}
          placeholder="Select categories"
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          {categories.length > 0 ? categories.map((c) => <Option key={c.id}>{c.slug}</Option>) : []}
        </Select>
      </Form.Item>

      <Form.Item
        label="Tag"
        name="tag_ids"
        rules={[
          {
            required: true,
            message: 'Please select atleast one tag!',
          },
        ]}
      >
        <Select
          mode="tags"
          style={{ width: '100%' }}
          placeholder="Select tags"
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          {tags.length > 0 ? tags.map((t) => <Option key={t.id}>{t.slug}</Option>) : []}
        </Select>
      </Form.Item>

      <Form.Item
        label="Status"
        name="status"
        rules={[
          {
            required: true,
            message: 'Please enter status!',
          },
        ]}
      >
        <Select
          showSearch
          style={{ width: '100%' }}
          placeholder="Select status"
          defaultValue={'Hide'}
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          <Option key={'Hide'}>Hide</Option>
          <Option key={'Unhide'}>Unhide</Option>
        </Select>
      </Form.Item>

      <Form.Item
        label="Type"
        name="product_type_id"
        rules={[
          {
            required: true,
            message: 'Please enter type!',
          },
        ]}
      >
        <Select
          showSearch
          style={{ width: '100%' }}
          placeholder="Select type"
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          {productType.length > 0
            ? productType.map((p) => <Option key={p.id}>{p.name}</Option>)
            : []}
        </Select>
      </Form.Item>

      <Form.Item wrapperCol={{ span: 12, offset: 6 }}>
        <Button type="primary" htmlType="submit">
          Submit
        </Button>
      </Form.Item>
    </Form>
  );
};

export default ProductCreate;
