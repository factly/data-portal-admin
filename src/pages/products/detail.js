import React from 'react';
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';

import ProductDetails from './components/ProductDetails';

const ProductDetail = () => {
  const { id } = useParams();
  const { product, currencies, tags } = useSelector(({ products, currencies, tags }) => ({
    currencies: currencies.items,
    tags: tags.items,
    product: products.items[id],
  }));

  const currencyCode = currencies[product.currency_id].iso_code;
  const productTags = [];
  product.tags.forEach((id) => productTags.push(tags[id]));

  return <ProductDetails product={product} productTags={productTags} currencyCode={currencyCode} />;
};

export default ProductDetail;
