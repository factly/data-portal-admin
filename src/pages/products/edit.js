import React, { useState } from 'react';
import { Form, Input, Button, notification, Select } from 'antd';
import { useParams } from 'react-router-dom';

const { Option } = Select;

const formItemLayout = {
  labelCol: { span: 6 },
  wrapperCol: { span: 14 },
};

const ProductEdit = (props) => {
  const [tags, setTags] = useState([]);
  const [categories, setCategories] = useState([]);
  const [currencies, setCurrencies] = useState([]);
  const [productType, setProductType] = useState([]);
  const [product, setProduct] = useState();

  const { id } = useParams();
  React.useEffect(() => {
    fetch(process.env.REACT_APP_API_URL + `/products/${id}`)
      .then((data) => data.json())
      .then((data) => {
        setProduct(data);
      })
      .catch((err) => {
        notification.error({
          message: 'Error',
          description: 'Something went wrong',
        });
      });
    fetch(process.env.REACT_APP_API_URL + '/tags')
      .then((data) => data.json())
      .then((data) => {
        setTags(data.nodes);
      });
    fetch(process.env.REACT_APP_API_URL + '/categories')
      .then((data) => data.json())
      .then((data) => {
        setCategories(data.nodes);
      });
    fetch(process.env.REACT_APP_API_URL + '/currencies')
      .then((data) => data.json())
      .then((data) => {
        setCurrencies(data.nodes);
      });
    fetch(process.env.REACT_APP_API_URL + '/types')
      .then((data) => data.json())
      .then((data) => {
        setProductType(data.nodes);
      });
  }, [id]);

  function handleChange(value) {
    console.log(`selected ${value}`);
  }

  const onFinish = (values) => {
    values.currency_id = parseInt(values.currency_id);
    values.product_type_id = parseInt(values.product_type_id);
    values.price = parseInt(values.price);
    values.category_ids = values.category_ids.map((id) => parseInt(id));
    values.tag_ids = values.tag_ids.map((id) => parseInt(id));

    fetch(process.env.REACT_APP_API_URL + `/products/${id}`, {
      method: 'PUT',
      body: JSON.stringify(values),
    })
      .then((res) => {
        if (res.ok) {
          return res.json();
        } else {
          throw new Error(res.status);
        }
      })
      .then((_) => {
        notification.success({
          message: 'Success',
          description: 'Product succesfully updated',
        });
        props.history.push('/products');
      })
      .catch((err) => {
        notification.error({
          message: 'Error',
          description: 'Something went wrong',
        });
      });
  };

  const initialValues = {
    slug: product ? product.slug : '',
    title: product ? product.title : '',
    price: product ? product.price : '',
    currency_id: product ? product.Currency.id : '',
    category_ids: product ? product.categories.map((category) => category.id) : [],
    tag_ids: product ? product.tags.map((tag) => tag.id) : [],
    status: product ? product.status : '',
    product_type_id: product ? product.ProductType.id : '',
  };

  return !product ? null : (
    <Form
      name="products_update"
      {...formItemLayout}
      initialValues={initialValues}
      onFinish={onFinish}
    >
      <Form.Item
        label="Title"
        name="title"
        rules={[
          {
            required: true,
            message: 'Please enter title!',
          },
        ]}
      >
        <Input placeholder="Ex. Crime In India" />
      </Form.Item>

      <Form.Item
        label="Slug"
        name="slug"
        rules={[
          {
            required: true,
            message: 'Please enter slug!',
          },
        ]}
      >
        <Input placeholder="Ex. crime-in-india" />
      </Form.Item>

      <Form.Item
        label="Currency"
        name="currency_id"
        rules={[
          {
            required: true,
            message: 'Please enter currency!',
          },
        ]}
      >
        <Select
          showSearch
          style={{ width: '100%' }}
          placeholder="Select currency"
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          {currencies.length > 0
            ? currencies.map((currency) => (
                <Select.Option key={currency.id} value={currency.id}>
                  {currency.name}
                </Select.Option>
              ))
            : []}
        </Select>
      </Form.Item>

      <Form.Item
        label="Price"
        name="price"
        rules={[
          {
            required: true,
            message: 'Please enter price!',
          },
        ]}
      >
        <Input placeholder="Ex. 1999" />
      </Form.Item>

      <Form.Item
        label="Category"
        name="category_ids"
        rules={[
          {
            required: true,
            message: 'Please select atleast one category!',
          },
        ]}
      >
        <Select
          mode="multiple"
          style={{ width: '100%' }}
          placeholder="Select categories"
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          {categories.length > 0
            ? categories.map((category) => (
                <Select.Option key={category.id} value={category.id}>
                  {category.title}
                </Select.Option>
              ))
            : []}
        </Select>
      </Form.Item>

      <Form.Item
        label="Tag"
        name="tag_ids"
        rules={[
          {
            required: true,
            message: 'Please select atleast one tag!',
          },
        ]}
      >
        <Select
          mode="multiple"
          style={{ width: '100%' }}
          placeholder="Select tags"
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          {tags.length > 0
            ? tags.map((tag) => (
                <Select.Option key={tag.id} value={tag.id}>
                  {tag.title}
                </Select.Option>
              ))
            : []}
        </Select>
      </Form.Item>

      <Form.Item
        label="Status"
        name="status"
        rules={[
          {
            required: true,
            message: 'Please enter status!',
          },
        ]}
      >
        <Select
          showSearch
          style={{ width: '100%' }}
          placeholder="Select status"
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          <Option key={1} value={'Show'}>
            Show
          </Option>
          <Option key={2} value={'Hide'}>
            Hide
          </Option>
        </Select>
      </Form.Item>

      <Form.Item
        label="Type"
        name="product_type_id"
        rules={[
          {
            required: true,
            message: 'Please enter type!',
          },
        ]}
      >
        <Select
          showSearch
          style={{ width: '100%' }}
          placeholder="Select type"
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          {productType.length > 0
            ? productType.map((product) => (
                <Select.Option key={product.id} value={product.id}>
                  {product.name}
                </Select.Option>
              ))
            : []}
        </Select>
      </Form.Item>

      <Form.Item wrapperCol={{ span: 12, offset: 6 }}>
        <Button type="primary" htmlType="submit">
          Submit
        </Button>
      </Form.Item>
    </Form>
  );
};

export default ProductEdit;
