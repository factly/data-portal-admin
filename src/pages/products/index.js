import React, { useState } from 'react';
import { Table, Input, Button, Popconfirm, Form, notification } from 'antd';
import moment from 'moment';
import { Link } from 'react-router-dom';
import { EditOutlined, DeleteOutlined } from '@ant-design/icons';

const EditableCell = ({ editing, dataIndex, title, record, index, children, ...restProps }) => {
  return (
    <td {...restProps}>
      {editing ? (
        <Form.Item
          name={dataIndex}
          style={{
            margin: 0,
          }}
          rules={[
            {
              required: true,
              message: `Please enter ${title}!`,
            },
          ]}
        >
          <Input />
        </Form.Item>
      ) : (
        children
      )}
    </td>
  );
};

const Products = (props) => {
  const [form] = Form.useForm();
  const [data, setData] = useState([]);
  const [total, setTotal] = useState(0);
  const [editingKey, setEditingKey] = useState('');

  React.useEffect(() => {
    fetch(process.env.REACT_APP_API_URL + '/products')
      .then((data) => data.json())
      .then((data) => {
        setData(data.nodes);
        setTotal(data.total);
      });
  }, []);

  const get = (page, limit) => {
    cancel();
    fetch(process.env.REACT_APP_API_URL + '/products?page=' + page + '&limit=' + limit)
      .then((data) => data.json())
      .then((data) => {
        setData(data.nodes);
        setTotal(data.total);
      });
  };

  const isEditing = (record) => record.id === editingKey;

  const cancel = () => {
    setEditingKey('');
  };

  const deleteProduct = (key) => {
    const index = data.findIndex((item) => item.id === key);
    if (index > -1) {
      fetch(process.env.REACT_APP_API_URL + '/products/' + key, {
        method: 'DELETE',
      })
        .then((res) => {
          if (res.status === 200) {
            const newData = [...data];
            newData.splice(index, 1);
            setData(newData);
            notification.success({
              message: 'Success',
              description: 'Product succesfully deleted',
            });
          }
        })
        .catch((err) => {
          notification.error({
            message: 'Error',
            description: 'Something went wrong',
          });
        });
    }
  };

  const columns = [
    {
      title: 'Title',
      dataIndex: 'title',
      width: '10%',
    },
    {
      title: 'Slug',
      dataIndex: 'slug',
      width: '10%',
    },
    {
      title: 'Currency',
      render: (record) => record.currency.iso_code,
      width: '10%',
    },
    {
      title: 'Price',
      dataIndex: 'price',
      width: '10%',
    },
    {
      title: 'Status',
      render: (record) => record.status,
      width: '10%',
    },
    {
      title: 'Categories',
      render: (record) => record.categories.map((c) => c.title).join(', '),
      width: '20%',
    },
    {
      title: 'Tags',
      render: (record) => record.tags.map((t) => t.title).join(', '),
      width: '20%',
    },
    {
      title: 'Created At',
      dataIndex: 'CreatedAt',
      width: '10%',
      render: (_, record) => {
        return <span title={record.CreatedAt}>{moment(record.CreatedAt).fromNow()}</span>;
      },
    },
    {
      title: 'Operation',
      dataIndex: 'operation',
      render: (_, record) => {
        return (
          <span>
            <Button
              type="primary"
              icon={<EditOutlined />}
              disabled={editingKey !== ''}
              onClick={() => {
                props.history.push(`/products/detail/${record.id}`);
              }}
              style={{
                marginRight: 8,
              }}
            >
              Edit
            </Button>
            <Popconfirm
              disabled={editingKey !== ''}
              title="Sure to delete?"
              onConfirm={() => deleteProduct(record.id)}
            >
              <Button disabled={editingKey !== ''} icon={<DeleteOutlined />}>
                Delete
              </Button>
            </Popconfirm>
          </span>
        );
      },
    },
  ];

  const mergedColumns = columns.map((col) => {
    if (!col.editable) {
      return col;
    }

    return {
      ...col,
      onCell: (record) => ({
        record,
        dataIndex: col.dataIndex,
        title: col.title,
        editing: isEditing(record),
      }),
    };
  });

  return (
    <div>
      <Link to={'/products/create'}>
        <Button type="primary" style={{ marginBottom: 16 }}>
          Add Product
        </Button>
      </Link>
      <Form form={form} component={false}>
        <Table
          components={{
            body: {
              cell: EditableCell,
            },
          }}
          bordered
          rowKey="id"
          dataSource={data}
          columns={mergedColumns}
          rowClassName="editable-row"
          pagination={{
            defaultPageSize: 5,
            onChange: get,
            total: total,
          }}
        />
      </Form>
    </div>
  );
};

export default Products;
