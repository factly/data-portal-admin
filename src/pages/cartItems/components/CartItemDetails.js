import React from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { Skeleton } from 'antd';

import ProductDetails from '../../products/components/ProductDetails';
import { getProduct } from '../../../actions/products';

const CartItemDetails = () => {
  const { id } = useParams();
  const dispatch = useDispatch();

  const { loading, product, currencies, tags } = useSelector(({ products, currencies, tags }) => ({
    loading: products.loading,
    product: products.items[id],
    currencies: currencies.items,
    tags: tags.items,
  }));

  React.useEffect(() => {
    dispatch(getProduct(id));
  }, []);

  if (loading) return <Skeleton />;

  const currencyCode = currencies[product.currency_id].iso_code;
  const productTags = [];
  product.tags.forEach((id) => productTags.push(tags[id]));

  return (
    <>
      <h2>Product:</h2>
      <ProductDetails product={product} currencyCode={currencyCode} productTags={productTags} />
    </>
  );
};

export default CartItemDetails;
