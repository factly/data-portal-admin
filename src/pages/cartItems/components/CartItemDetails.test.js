import React from 'react';
import { useDispatch, Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { act } from '@testing-library/react';

import '../../../matchMedia.mock';
import CartItemDetails from './CartItemDetails';
import { getProduct } from '../../../actions/products';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));
jest.mock('../../../actions/products', () => ({
  getProduct: jest.fn(),
}));
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn().mockReturnValue({ id: '1' }),
}));

describe('Cart Item Details component', () => {
  let store;
  let mockedDispatch;
  describe('snapshot testing', () => {
    beforeEach(() => {
      store = mockStore({
        cartItems: {
          loading: false,
          ids: [1],
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1],
            },
          ],
          items: {
            1: {
              id: 1,
              user_id: 1,
              status: 'status',
              product_id: 1,
              created_at: '2020-12-12',
            },
          },
          total: 1,
        },
        products: {
          loading: false,
          ids: [1],
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1],
            },
          ],
          items: {
            1: {
              id: 1,
              title: 'Product',
              currency_id: 1,
              tags: [1],
            },
          },
          total: 1,
        },
        currencies: {
          loading: false,
          ids: [1],
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1],
            },
          ],
          items: {
            1: {
              id: 1,
              name: 'Indian Rupee',
              iso_code: 'INR',
            },
          },
        },
        tags: {
          loading: false,
          ids: [1],
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1],
            },
          ],
          items: {
            1: {
              id: 1,
              title: 'Tag',
            },
          },
          total: 1,
        },
      });
      store.dispatch = jest.fn(() => ({}));
      mockedDispatch = jest.fn(() => Promise.resolve({}));
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should render the component', () => {
      const tree = mount(
        <Provider store={store}>
          <CartItemDetails></CartItemDetails>
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
    it('should render the component when loading', () => {
      store = mockStore({
        cartItems: {
          loading: false,
          ids: [1],
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1],
            },
          ],
          items: {
            1: {
              id: 1,
              user_id: 1,
              status: 'status',
              product_id: 1,
              created_at: '2020-12-12',
            },
          },
          total: 1,
        },
        products: {
          loading: true,
          ids: [1],
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1],
            },
          ],
          items: {
            1: {
              id: 1,
              title: 'Product',
              currency_id: 1,
              tags: [1],
            },
          },
          total: 1,
        },
        currencies: {
          loading: false,
          ids: [],
          req: [],
          items: {},
        },
        tags: {
          loading: false,
          ids: [],
          req: [],
          items: {},
          total: 0,
        },
      });
      const tree = mount(
        <Provider store={store}>
          <CartItemDetails></CartItemDetails>
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper;
    beforeEach(() => {
      store = mockStore({
        cartItems: {
          loading: false,
          ids: [1],
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1],
            },
          ],
          items: {
            1: {
              id: 1,
              user_id: 1,
              status: 'status',
              product_id: 1,
              created_at: '2020-12-12',
            },
          },
          total: 1,
        },
        products: {
          loading: false,
          ids: [1],
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1],
            },
          ],
          items: {
            1: {
              id: 1,
              title: 'Product',
              currency_id: 1,
              tags: [1],
            },
          },
          total: 1,
        },
        currencies: {
          loading: false,
          ids: [1],
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1],
            },
          ],
          items: {
            1: {
              id: 1,
              name: 'Indian Rupee',
              iso_code: 'INR',
            },
          },
        },
        tags: {
          loading: false,
          ids: [1],
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1],
            },
          ],
          items: {
            1: {
              id: 1,
              title: 'Tag',
            },
          },
          total: 1,
        },
      });
      store.dispatch = jest.fn(() => ({}));
      mockedDispatch = jest.fn(() => Promise.resolve({}));
      useDispatch.mockReturnValue(mockedDispatch);
    });
    afterEach(() => {
      wrapper.unmount();
    });

    it('should call getProduct action', () => {
      getProduct.mockReset();
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <CartItemDetails />
          </Provider>,
        );
      });
      expect(getProduct).toHaveBeenCalledWith(1);
    });
  });
});
