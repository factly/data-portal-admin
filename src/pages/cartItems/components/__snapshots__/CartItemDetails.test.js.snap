// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Item Details component snapshot testing should render the component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CartItemDetails>
    <h2>
      Product:
    </h2>
    <ProductDetails
      currencyCode="INR"
      product={
        Object {
          "currency_id": 1,
          "id": 1,
          "tags": Array [
            1,
          ],
          "title": "Product",
        }
      }
      productTags={
        Array [
          Object {
            "id": 1,
            "title": "Tag",
          },
        ]
      }
    >
      <Card>
        <div
          className="ant-card ant-card-bordered"
        >
          <div
            className="ant-card-body"
            style={Object {}}
          >
            <Descriptions
              column={1}
              title="Product"
            >
              <div
                className="ant-descriptions"
              >
                <div
                  className="ant-descriptions-header"
                >
                  <div
                    className="ant-descriptions-title"
                  >
                    Product
                  </div>
                </div>
                <div
                  className="ant-descriptions-view"
                >
                  <table>
                    <tbody>
                      <Row
                        colon={true}
                        index={0}
                        key="0"
                        prefixCls="ant-descriptions"
                        row={
                          Array [
                            <DescriptionsItem
                              label="Slug"
                            />,
                          ]
                        }
                        vertical={false}
                      >
                        <tr
                          className="ant-descriptions-row"
                          key="0"
                        >
                          <Cell
                            colon={true}
                            component="td"
                            itemPrefixCls="ant-descriptions"
                            key="item-0"
                            label="Slug"
                            span={1}
                          >
                            <td
                              className="ant-descriptions-item"
                              colSpan={1}
                            >
                              <div
                                className="ant-descriptions-item-container"
                              >
                                <span
                                  className="ant-descriptions-item-label"
                                >
                                  Slug
                                </span>
                              </div>
                            </td>
                          </Cell>
                        </tr>
                      </Row>
                      <Row
                        colon={true}
                        index={1}
                        key="1"
                        prefixCls="ant-descriptions"
                        row={
                          Array [
                            <DescriptionsItem
                              label="Price"
                            >
                               
                              INR
                            </DescriptionsItem>,
                          ]
                        }
                        vertical={false}
                      >
                        <tr
                          className="ant-descriptions-row"
                          key="1"
                        >
                          <Cell
                            colon={true}
                            component="td"
                            content={
                              Array [
                                undefined,
                                " ",
                                "INR",
                              ]
                            }
                            itemPrefixCls="ant-descriptions"
                            key="item-0"
                            label="Price"
                            span={1}
                          >
                            <td
                              className="ant-descriptions-item"
                              colSpan={1}
                            >
                              <div
                                className="ant-descriptions-item-container"
                              >
                                <span
                                  className="ant-descriptions-item-label"
                                >
                                  Price
                                </span>
                                <span
                                  className="ant-descriptions-item-content"
                                >
                                   
                                  INR
                                </span>
                              </div>
                            </td>
                          </Cell>
                        </tr>
                      </Row>
                      <Row
                        colon={true}
                        index={2}
                        key="2"
                        prefixCls="ant-descriptions"
                        row={
                          Array [
                            <DescriptionsItem
                              label="Status"
                            />,
                          ]
                        }
                        vertical={false}
                      >
                        <tr
                          className="ant-descriptions-row"
                          key="2"
                        >
                          <Cell
                            colon={true}
                            component="td"
                            itemPrefixCls="ant-descriptions"
                            key="item-0"
                            label="Status"
                            span={1}
                          >
                            <td
                              className="ant-descriptions-item"
                              colSpan={1}
                            >
                              <div
                                className="ant-descriptions-item-container"
                              >
                                <span
                                  className="ant-descriptions-item-label"
                                >
                                  Status
                                </span>
                              </div>
                            </td>
                          </Cell>
                        </tr>
                      </Row>
                      <Row
                        colon={true}
                        index={3}
                        key="3"
                        prefixCls="ant-descriptions"
                        row={
                          Array [
                            <DescriptionsItem
                              label="Tags"
                            >
                              <ForwardRef(InternalTag)>
                                Tag
                              </ForwardRef(InternalTag)>
                            </DescriptionsItem>,
                          ]
                        }
                        vertical={false}
                      >
                        <tr
                          className="ant-descriptions-row"
                          key="3"
                        >
                          <Cell
                            colon={true}
                            component="td"
                            content={
                              Array [
                                <ForwardRef(InternalTag)>
                                  Tag
                                </ForwardRef(InternalTag)>,
                              ]
                            }
                            itemPrefixCls="ant-descriptions"
                            key="item-0"
                            label="Tags"
                            span={1}
                          >
                            <td
                              className="ant-descriptions-item"
                              colSpan={1}
                            >
                              <div
                                className="ant-descriptions-item-container"
                              >
                                <span
                                  className="ant-descriptions-item-label"
                                >
                                  Tags
                                </span>
                                <span
                                  className="ant-descriptions-item-content"
                                >
                                  <Tag
                                    key="1"
                                  >
                                    <span
                                      className="ant-tag"
                                      style={
                                        Object {
                                          "backgroundColor": undefined,
                                        }
                                      }
                                    >
                                      Tag
                                    </span>
                                  </Tag>
                                </span>
                              </div>
                            </td>
                          </Cell>
                        </tr>
                      </Row>
                      <Row
                        colon={true}
                        index={4}
                        key="4"
                        prefixCls="ant-descriptions"
                        row={
                          Array [
                            <DescriptionsItem
                              label="Created At"
                              span={1}
                            >
                              a few seconds ago
                            </DescriptionsItem>,
                          ]
                        }
                        vertical={false}
                      >
                        <tr
                          className="ant-descriptions-row"
                          key="4"
                        >
                          <Cell
                            colon={true}
                            component="td"
                            content="a few seconds ago"
                            itemPrefixCls="ant-descriptions"
                            key="item-0"
                            label="Created At"
                            span={1}
                          >
                            <td
                              className="ant-descriptions-item"
                              colSpan={1}
                            >
                              <div
                                className="ant-descriptions-item-container"
                              >
                                <span
                                  className="ant-descriptions-item-label"
                                >
                                  Created At
                                </span>
                                <span
                                  className="ant-descriptions-item-content"
                                >
                                  a few seconds ago
                                </span>
                              </div>
                            </td>
                          </Cell>
                        </tr>
                      </Row>
                    </tbody>
                  </table>
                </div>
              </div>
            </Descriptions>
          </div>
        </div>
      </Card>
    </ProductDetails>
  </CartItemDetails>
</Provider>
`;

exports[`Cart Item Details component snapshot testing should render the component when loading 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CartItemDetails>
    <Skeleton
      avatar={false}
      paragraph={true}
      title={true}
    >
      <div
        className="ant-skeleton"
      >
        <div
          className="ant-skeleton-content"
        >
          <Title
            prefixCls="ant-skeleton-title"
            width="38%"
          >
            <h3
              className="ant-skeleton-title"
              style={
                Object {
                  "width": "38%",
                }
              }
            />
          </Title>
          <Paragraph
            prefixCls="ant-skeleton-paragraph"
            rows={3}
            width="61%"
          >
            <ul
              className="ant-skeleton-paragraph"
            >
              <li
                key="0"
                style={
                  Object {
                    "width": undefined,
                  }
                }
              />
              <li
                key="1"
                style={
                  Object {
                    "width": undefined,
                  }
                }
              />
              <li
                key="2"
                style={
                  Object {
                    "width": "61%",
                  }
                }
              />
            </ul>
          </Paragraph>
        </div>
      </div>
    </Skeleton>
  </CartItemDetails>
</Provider>
`;
